Mathlib/RingTheory/Artinian.lean

```diff
@@ -490,16 +490,15 @@ variable (S : Submonoid R) (L : Type*) [CommRing L] [Algebra R L] [IsLocalizatio
 theorem localization_surjective : Function.Surjective (algebraMap R L) := by
   intro r'
   obtain ⟨r₁, s, rfl⟩ := IsLocalization.mk'_surjective S r'
-  obtain ⟨r₂, h⟩ : ∃ r : R, IsLocalization.mk' L 1 s = algebraMap R L r
-  swap
-  · exact ⟨r₁ * r₂, by rw [IsLocalization.mk'_eq_mul_mk'_one, map_mul, h]⟩
-  obtain ⟨n, r, hr⟩ := IsArtinian.exists_pow_succ_smul_dvd (s : R) (1 : R)
-  use r
-  rw [smul_eq_mul, smul_eq_mul, pow_succ, mul_assoc] at hr
-  apply_fun algebraMap R L at hr
-  simp only [map_mul] at hr
-  rw [← IsLocalization.mk'_one (M := S) L, IsLocalization.mk'_eq_iff_eq, mul_one, Submonoid.coe_one,
-    ← (IsLocalization.map_units L (s ^ n)).mul_left_cancel hr, map_mul]
+  obtain ⟨r₂, h⟩ : ∃ r : R, IsLocalization.mk' L 1 s = algebraMap R L r := by
+    obtain ⟨n, r, hr⟩ := IsArtinian.exists_pow_succ_smul_dvd (s : R) (1 : R)
+    use r
+    rw [smul_eq_mul, smul_eq_mul, pow_succ, mul_assoc] at hr
+    apply_fun algebraMap R L at hr
+    simp only [map_mul] at hr
+    rw [← IsLocalization.mk'_one (M := S) L, IsLocalization.mk'_eq_iff_eq, mul_one,
+      Submonoid.coe_one, ← (IsLocalization.map_units L (s ^ n)).mul_left_cancel hr, map_mul]
+  exact ⟨r₁ * r₂, by rw [IsLocalization.mk'_eq_mul_mk'_one, map_mul, h]⟩
```

Comment:
This really does not look like an improvement. That's why there was a `swap`

Reply:
I have put (on the other PR) a TODO, investigating using `rsuffices`.

Reply:
I have put (on the other PR) a TODO, investigating using `rsuffices`.